### ISSUE TITLE
Local Alignment Near Reference Sequence Ends

### ISSUE THREAD

**Issue by DarioS** [BenLangmead/bowtie2#106]
Reads which have a relatively short sequence that does not map to a reference sequence and a relatively long sequence which maps perfectly to one end of a reference sequence aren't mapped. Is there any way to do this with `bowtie2` using `--local`? I have a task which involves ranking the references (transcripts) by the percentage of bases covered by HTS reads, and an important proportion of uncovered bases sometimes happens at both ends of the reference. The chances of mapping at the edge of a reference sequence are lower than mapping internally, particularly if the average coverage of the reference is low, so I'm hoping to make it more equal somehow.

**Comment by BenLangmead** [comment]
Yes, this should be doable with `--local` mode.  Unaligned bases will be soft clipped.  To tune how long the match may be relative to the number of soft-clipped bases, you'll have to adjust the scoring scheme.  See `--score-min` and `--ma` in particular.

**Comment by DarioS** [comment]
My thinking is that the options `--local --ma 0 --mp 1,1 --rdg 0,1 --rfg 0,1 --score-min C,0` should work for this scenario but they don't. I think that it would give a read with CIGAR 20S80M and a second with 100M both an alignment score of 0 and report them, but it doesn't.

In the section of the user manual titled "Local alignment score example" could you add another example which explicitly shows a read mapping to the reference with some amount of soft-clipping in case the ambiguity caused me to misunderstand the scoring scheme of Bowtie 2?

**Comment by DarioS** [comment]
I just tried mapping a read with a reference sequence overhang using `bwa mem` and it works as intended, soft-clipping the few bases protruding beyond the reference sequence. `bowtie2` might need fixing.

**Comment by BenLangmead** [comment]
Can you clarify what we should be doing to address this issue?

**Comment by durrantmm** [comment]
I think I may be able to clarify. I ran a test, and in --local mode, reverse oriented reads where the forward pair is unmapped are not softclipped on their left side when they should be. This is what I did:

- I study insertion sequences. I simulated a 1500 bp insertion sequence into a 10kb randomly generated genome.
- I created 500 bp fragments at each base in the genome, with 100 bp paired-ends.
- I then aligned to the genome without the insertion using bowtie2 and viewed results in IGV. 
- Reads that overlapped the insertion site were softclipped, and the softclipped end could indicate which side of the insertion sequence the read came from.
- I expected to see coverage drop evenly on both the left and the right side of the insertion sequence, as reads could no longer map with accuracy as the size of the softclip increases.
- Coverage was not even on both sides of the insertion sequence. Reverse oriented reads were not soft clipped that did not have a mapped pair (because it was derived from the absent insertion sequence) were never softclipped on their left side, while their forward-oriented counterparts were properly softclipped.

Here are some images:

I also simulated a direct repeat, which is the pillar in the middle. You can see the uneven changes in coverage.
<img width="1244" alt="screen shot 2018-01-07 at 9 18 59 pm" src="https://user-images.githubusercontent.com/10098419/34659684-6ecab250-f3f0-11e7-9e5f-0b059ef8a6bf.png">

Here, the reads are grouped by whether or not their pair mapped.You can see on the bottom panel that reverse reads with unmapped pairs are never softclipped as they should be.
![image](https://user-images.githubusercontent.com/10098419/34659692-9e2dee86-f3f0-11e7-804c-5fe5f5d09a54.png)

I hope that makes sense.

**Comment by CarloscostaH** [comment]
Hi @durrantmm ! So as I can see, you're having some issues with Bowtie, I had the same problem when I was mapping mitochondrial genomes from distant species of flies. What I did was to change some parameters in Bowtie2 in order to make things more sensitive, in that case I used: 

--very-sensitive-local -N 1 -i S,0.5,0.2

The parameters may change from one project to another, but I highly suggest you to give a try and see how does coverage behave on your data. Another thing is to use --end-to-end mode, which, differently from local alignment, puts gaps '-' in the reference instead of inside your reads. What you might have experienced were the mapping 'forcing' your reads to fit inside your artificial insert. I have a question, how  many coverage did you obtained over this region? By seing the picture I can guess that it was 14X for Bowtie and 3X~ for BWA.
